iib/workers/tasks/legacy.py

```diff
@@ -51,19 +51,22 @@ def _get_base_dir_and_pkg_name(package_dir):
     return os.path.dirname(package_dir), os.path.basename(package_dir)
 
 
-def get_legacy_support_packages(bundles):
+def get_legacy_support_packages(bundles, force_backport=False):
     """
     Get the packages that must be pushed to the legacy application registry.
 
     :param list<str> bundles: a list of strings representing the pull specifications of the bundles
         to add to the index image being built.
+    :param bool force_backport: if True, backport legacy support is forced for every package
     :return: a set of packages that require legacy support
     :rtype: set
     """
+    if force_backport:
+        log.info('Backport legacy support will be forced')
```

Comment:
Do you think it might be helpful to have the state_reason should that the backport legacy support is being forced?

Reply:
Hmm that's a good point. Otherwise, this information would only be available in the logs. Should I do a new state, or maybe add a new parameter to build request representation (DB change)?

I think a state_reason should be enough.

Reply:
Done.